dev/reference/tools/sync-diff-inspector/sync-diff-inspector.md

```diff
@@ -0,0 +1,214 @@
+
+## sync-diff-inspector 的使用
+
+### 使用限制
+
+* 目前不支持在线校验，需要保证上下游校验的表中没有数据写入，或者保证某个范围内的数据不再变更，通过配置 `range` 来校验这个范围内的数据。
+
+* 不支持 JSON、BIT、BINARY、BLOB 等类型的数据，在校验时需要设置 `ignore-columns` 忽略检查这些类型的数据。
+
+* FLOAT、DOUBLE 等浮点数类型在 TiDB 和 MySQL 中的实现方式不同，在计算 checksum 时可能存在差异，如果发现因为这些类型的数据导致的数据校验不一致，需要设置 `ignore-columns` 忽略这些列的检查。
+
+### 数据库权限
+
+sync-diff-inspector 需要获取表结构信息、查询数据、建 checkpoint 库保存断点信息，需要的数据库权限如下：
+
+- 上游数据库
+
+    SELECT（查数据进行对比）
+
+    SHOW_DATABASES (查看库名)
+
+    RELOAD (查看表结构)
+
+- 下游数据库
+  
+    SELECT （查数据进行对比）
+
+    CREATE （创建 checkpoint 库和表）
+
+    DELETE （删除 checkpoint 表中的信息）
+
+    INSERT  （写入 checkpoint 表）
+
+    UPDATE（修改 checkpoint 表）
+
+    SHOW_DATABASES (查看库名)
+
+    RELOAD (查看表结构)
+
+### 通用配置文件说明
+
+sync-diff-inspector 的配置总共分为三个部分：
+
+- Global config: 通用配置，包括日志级别、划分 chunk 的大小、校验的线程数量等。
+- Tables config: 配置校验哪些表，如果有的表在上下游有一定的映射关系或者有一些特殊要求，则需要对指定的表进行配置。
+- Databases config: 配置上下游数据库实例。
+
+下面是一个完整配置文件的说明：
+
+``` toml
+# Diff Configuration.
+
+######################### Global config #########################
+
+# 日志级别，可以设置为 info、debug
+log-level = "info"
+
+# sync-diff-inspector 根据主键／唯一键／索引将数据划分为多个 chunk，
+# 对每一个 chunk 的数据进行对比。使用 chunk-size 设置 chunk 的大小
+chunk-size = 1000
+
+# 检查数据的线程数量
+check-thread-count = 4
+
+# 抽样检查的比例，如果设置为 100 则检查全部数据
+sample-percent = 100
+
+# 通过计算 chunk 的 checksum 来对比数据，如果不开启则逐行对比数据
+use-checksum = true
+
+# 如果设置为 true 则只会通过计算 checksum 来校验数据，如果上下游的 checksum 不一致也不会查出数据再进行校验
+only-use-checksum = false
+
+# 是否使用上次校验的 checkpoint，如果开启，则只校验上次未校验以及校验失败的 chunk
+use-checkpoint = true
+
+# 不对比数据
+ignore-data-check = false
+
+# 不对比表结构
+ignore-struct-check = false
+
+# 保存用于修复数据的 sql 的文件名称
+fix-sql-file = "fix.sql"
+
+######################### Tables config #########################
+
+# 如果需要对比大量的不同库名或者表名的表的数据，可以通过 table-rule 来设置映射关系。可以只配置 schema 或者 table 的映射关系，也可以都配置
+#[[table-rules]]
+    # schema-pattern 和 table-pattern 支持通配符 *?
+    #schema-pattern = "test_*"
+    #table-pattern = "t_*"
+    #target-schema = "test"
+    #target-table = "t"
+
+# 配置需要对比的*目标数据库*中的表
+[[check-tables]]
+    # 目标库中数据库的名称
+    schema = "test"
+
+    # 需要检查的表
+    tables = ["test1", "test2", "test3"]
+
+    # 支持使用正则表达式配置检查的表，需要以‘~’开始，
+    # 下面的配置会检查所有表名以‘test’为前缀的表
+    # tables = ["~^test.*"]
+    # 下面的配置会检查配置库中所有的表
+    # tables = ["~^"]
+
+# 对部分表进行特殊的配置，配置的表必须包含在 check-tables 中
+[[table-config]]
+    # 目标库中数据库的名称
+    schema = "test"
+
+    # 表名
+    table = "test3"
+
+    # 指定用于划分 chunk 的列，如果不配置该项，sync-diff-inspector 会选取一个合适的列（主键／唯一键／索引）
+    index-field = "id"
+
+    # 指定检查的数据的范围，需要符合 sql 中 where 条件的语法
+    range = "age > 10 AND age < 20"
+
+    # 如果是对比多个分表与总表的数据，则设置为 true
+    is-sharding = false
+
+    # 在某些情况下字符类型的数据的排序会不一致，通过指定 collation 来保证排序的一致，
+    # 需要与数据库中 charset 的设置相对应
+    # collation = "latin1_bin"
+
+    # 忽略某些列的检查，例如 sync-diff-inspector 目前还不支持的一些类型（json，bit，blob 等），
+    # 或者是浮点类型数据在 TiDB 和 MySQL 中的表现可能存在差异，可以使用 ignore-columns 忽略检查这些列
+    # ignore-columns = ["name"]
```

Comment:
missing `remove-columns` config, is it needed to add some comparison between `ignore-columns` and `remove-columns`

Reply:
https://github.com/pingcap/tidb-tools/pull/248
in this pr, I will delete the `remove-columns` config